LiveSplit.TR3.asl

```diff
@@ -144,19 +144,32 @@ reset
 
 split
 {
-    // Determine if a split should occur.
     bool shouldSplit;
-    if (current.level == 19)
+    if (current.level == 19 && old.level == 19)
+    {
+        // While playing the "last" level (regardless of IL/FG and if the bonus level, All Hallows, will be played)
+        // the timer should split when the player's control ends and the helicopter scene begins.
+        // This occurs when the player hits the end level trigger which sets levelComplete to true.
         shouldSplit = current.levelComplete && !old.levelComplete;
+    }
     else if (settings["IL"])
+    {
+        // For IL timing, the timer should stop when the end-level stats screen appears.
+        // Checking levelComplete ensures this check works in all cases.
         shouldSplit = current.levelComplete && current.isStatsScreen && !old.isStatsScreen;
```

Comment:
To do: try this without `current.levelComplete`, if works, then I think the comments are not needed.

Reply:
I did not try without the levelComplete check but I removed the comment.